service/history/historyReplicator.go

```diff
@@ -185,8 +154,160 @@ func (r *historyReplicator) ApplyEvents(request *h.ReplicateEventsRequest) error
 				return err
 			}
 
+		case shared.EventTypeActivityTaskFailed:
+			msBuilder.ReplicateActivityTaskFailedEvent(event)
+
+		case shared.EventTypeActivityTaskTimedOut:
+			msBuilder.ReplicateActivityTaskTimedOutEvent(event)
+
+		case shared.EventTypeActivityTaskCancelRequested:
+			msBuilder.ReplicateActivityTaskCancelRequestedEvent(event)
+
+		case shared.EventTypeActivityTaskCanceled:
+			msBuilder.ReplicateActivityTaskCanceledEvent(event)
+
+		case shared.EventTypeRequestCancelActivityTaskFailed:
+			// No mutable state action is needed
+
+		case shared.EventTypeTimerStarted:
+			msBuilder.ReplicateTimerStartedEvent(event)
+
+		case shared.EventTypeTimerFired:
+			msBuilder.ReplicateTimerFiredEvent(event)
+
+		case shared.EventTypeTimerCanceled:
+			msBuilder.ReplicateTimerCanceledEvent(event)
+
+		case shared.EventTypeCancelTimerFailed:
+			// No mutable state action is needed
```

Comment:
not sure I understand it. But I thought we need to replicate every events, is that not the case?